server/channels/app/job.go

```diff
@@ -75,42 +75,38 @@ func (a *App) GetJobsByTypesAndStatuses(rctx request.CTX, jobTypes []string, sta
 }
 
 func (a *App) CreateJob(rctx request.CTX, job *model.Job) (*model.Job, *model.AppError) {
-	switch job.Type {
-	case model.JobTypeAccessControlSync:
-		// Route ABAC jobs to specialized deduplication handler
-		return a.CreateAccessControlSyncJob(rctx, job.Data)
-	default:
-		return a.Srv().Jobs.CreateJob(rctx, job.Type, job.Data)
-	}
+	return a.Srv().Jobs.CreateJob(rctx, job.Type, job.Data)
 }
 
 func (a *App) CreateAccessControlSyncJob(rctx request.CTX, jobData map[string]string) (*model.Job, *model.AppError) {
 	// Get the policy_id (channel ID) from job data to scope the deduplication
 	policyID, exists := jobData["policy_id"]
+	if !exists || policyID == "" {
+		return nil, model.NewAppError("CreateAccessControlSyncJob", "app.job.create_access_control_sync.missing_policy_id", nil, "", http.StatusBadRequest)
+	}
 
-	// If policy_id is provided, this is a channel-specific job that needs deduplication
-	if exists && policyID != "" {
-		// Find existing pending or in-progress jobs for this specific policy/channel
-		existingJobs, err := a.Srv().Store().Job().GetByTypeAndData(rctx, model.JobTypeAccessControlSync, map[string]string{
-			"policy_id": policyID,
-		}, true, model.JobStatusPending, model.JobStatusInProgress)
-		if err != nil {
-			return nil, model.NewAppError("CreateAccessControlSyncJob", "app.job.get_existing_jobs.error", nil, "", http.StatusInternalServerError).Wrap(err)
-		}
+	// Find existing pending or in-progress jobs for this specific policy/channel
+	existingJobs, err := a.Srv().Store().Job().GetByTypeAndData(rctx, model.JobTypeAccessControlSync, map[string]string{
+		"policy_id": policyID,
+	})
+	if err != nil {
+		return nil, model.NewAppError("CreateAccessControlSyncJob", "app.job.get_existing_jobs.error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
 
-		// Cancel any existing active jobs for this policy (all returned jobs are already active)
-		for _, job := range existingJobs {
+	// Cancel any existing pending or in-progress jobs for this policy
+	for _, job := range existingJobs {
```

Comment:
Will ever this list be too long? I guess we keep indefinitely the success jobs, so this may get long.

Not sure if it will be ever "too long" though.

If we wanted to avoid this, we could let the database handle it, by filtering in the database query by Status.  But 0/5.